test/Storage/MessageTest.php

```diff
@@ -415,6 +416,35 @@ public function testMultivalueToHeader()
         $this->assertEquals('nicpoń', $addressList->get('bar@example.pl')->getName());
     }
 
+    /**
+     * Test case for creating mail with Mail\Message containing \n\n,
+     * serializing it to text and loading with Storage\Message.
+     *
+     * Testcase showing that if mail is created with text body
+     * that is \n separated, loading such mail with Storage\Message
+     * causes parse error because it goes into mail body to find headers.
+     */
+    public function testDogFood() {
+        $text = "Hello, bödi tekst\n\nBye";
+
+        $part = new Mime\Part($text);
+        $part->type = 'text/plain';
+        $part->charset = 'UTF-8';
+
+        $mime = new Mime\Message();
+        $mime->addPart($part);
+
+        $message = new Mail\Message();
+        $message->setEncoding('UTF-8');
+        $message->setBody($text);
+
+        $raw = $message->toString();
+        $this->assertContains($text, $raw, "original text is preserved");
+
+        $message = new Storage\Message(['raw' => $raw]);
```

Comment:
I think this is a `Zend\Mime\Decode::splitMessage()` issue.

`Zend\Mail\Storage\Part` calls on that method to split the message into headers and content. My suspicion is that it's searching until the _last_ occurence of the `\n\n` sequence, instead of the _first_.

Can you please try and create and post a reproduce case against the zend-mime component? Thanks in advance!